lib/kitchen/driver/ec2.rb

```diff
@@ -354,6 +361,8 @@ def hostname(server, interface_type = nil)
           potential_hostname = nil
           INTERFACE_TYPES.values.each do |type|
             potential_hostname ||= server.send(type)
+            # AWS returns an empty string if the dns name isn't populated yet
```

Comment:
+1 I was thinking of this soley from inside a VPC, good call. Little short-sighted on my part.